pkg/dashboard/apiserver.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dashboard
+
+import (
+	"net/http"
+
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/apiserver"
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/config"
+	"github.com/pingcap-incubator/tidb-dashboard/pkg/keyvisual/region"
+	"go.etcd.io/etcd/clientv3"
+
+	"github.com/pingcap/pd/v4/pkg/dashboard/keyvisual/input"
+	"github.com/pingcap/pd/v4/server"
+)
+
+func newAPIService(srv *server.Server) (*apiserver.Service, error) {
+	cfg := srv.GetConfig()
+
+	etcdCfg, err := cfg.GenEmbedEtcdConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	dashboardCfg := &config.Config{
+		DataDir:    cfg.DataDir,
+		PDEndPoint: etcdCfg.ACUrls[0].String(),
+	}
+	dashboardCfg.ClusterTLSConfig, err = cfg.Security.ToTLSConfig()
```

Comment:
How do we get TiDBTLSConfig now?

Reply:
See https://github.com/pingcap/pd/pull/2246

Reply:
Looks like it hasn't merged yet